app/src/main/java/me/guillaumin/android/osmtracker/gpx/TrackFileProvider.java

```diff
@@ -0,0 +1,121 @@
+package me.guillaumin.android.osmtracker.gpx;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.database.AbstractCursor;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.ParcelFileDescriptor;
+import android.util.Log;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+
+/**
```

Comment:
Standard mail app calls the query() method (don't know why).

> it may lead into hard to diagnose issue

perhaps that's the case with that "file too large..." problem with standard mail app. It works on my emulator, but it shows the same error on my real device (I've used gmail for sending my tracks before). Need some research.